src/COWShedResolver.sol

```diff
@@ -0,0 +1,49 @@
+import {
+    INameResolver, IReverseRegistrar, IENS, IAddrResolver, ENS, ADDR_REVERSE_NODE, sha3HexAddress
+} from "./ens.sol";
+import { LibString } from "solady/utils/LibString.sol";
+
+abstract contract COWShedResolver is INameResolver, IAddrResolver {
+    mapping(bytes32 => address) public reverseResolutionNodeToAddress;
+    mapping(bytes32 => address) public forwardResolutionNodeToAddress;
+    bytes32 immutable baseName_;
+    bytes32 immutable baseNode;
```

Comment:
nit: consistent style with use of underscores (no underscores in state variables, underscores in local stack / memory vars)